core/service/src/client/mod.rs

```diff
@@ -6040,6 +6040,310 @@ pub(crate) mod tests {
+
+    #[cfg(feature = "insecure")]
+    #[tokio::test(flavor = "multi_thread")]
+    #[serial]
+    async fn default_insecure_crs_backup() {
+        let amount_parties = 4;
+        let param = FheParameter::Default;
+        let dkg_param: WrappedDKGParams = param.into();
+
+        let req_id: RequestId = derive_request_id(&format!(
+            "default_insecure_crs_backup{amount_parties}_{param:?}",
+        ))
+        .unwrap();
+        let test_path = None;
+        purge(test_path, test_path, test_path, &req_id, amount_parties).await;
+        let (_kms_servers, kms_clients, internal_client) =
+            threshold_handles(*dkg_param, amount_parties, true, None, None).await;
+        // Purge the backup after booting the server to ensure nothing is backeup automatically during boot
+        purge_backup(test_path, amount_parties).await;
+
+        run_crs(
+            param,
+            &kms_clients,
+            &internal_client,
+            true, // insecure execution
+            &req_id,
+            Some(16),
+        )
+        .await;
+
+        // Generated crs, delete it from private storage
+        for i in 1..=amount_parties {
+            let mut priv_storage: FileStorage =
+                FileStorage::new(None, StorageType::PRIV, Some(Role::indexed_from_one(i))).unwrap();
+            delete_all_at_request_id(&mut priv_storage, &req_id).await;
+        }
+
+        // Make sure file is actually deleted
+        tokio::time::sleep(tokio::time::Duration::from_millis(300)).await;
+
+        // Now try to restore
+        let mut resp_tasks = JoinSet::new();
+        for i in 1..=amount_parties as u32 {
+            let mut cur_client = kms_clients.get(&i).unwrap().clone();
+            resp_tasks.spawn(async move {
+                let req = Empty {};
+                // send query
+                cur_client
+                    .custodian_backup_restore(tonic::Request::new(req))
+                    .await
+            });
+        }
+        while let Some(res) = resp_tasks.join_next().await {
+            match res {
+                Ok(res) => match res {
+                    Ok(resp) => {
+                        println!("Custodian backup restore response: {resp:?}");
+                    }
+                    Err(e) => {
+                        panic!("Error while restoring: {e}");
+                    }
+                },
+                Err(e) => {
+                    panic!("Error while joining threads: {e}");
+                }
+            }
+        }
+
+        // Trying to restore again will fail since the file is already restored
+        let mut resp_tasks = JoinSet::new();
+        for i in 1..=amount_parties as u32 {
+            let mut cur_client = kms_clients.get(&i).unwrap().clone();
+            resp_tasks.spawn(async move {
+                let req = Empty {};
+                // send query
+                cur_client
+                    .custodian_backup_restore(tonic::Request::new(req))
```

Comment:
unlike the other GRPC calls, this is a blocking operation. I suppose it's ok like this since it's not something that the connector will use, at least not now.

Reply:
I think it is fine. As I believe the code will be removed again in a couple of weeks once the proper backup is written. I also don't think we ever want to execute this kind of code from the connector.